### PR TITLE
feat(intersection): add conflicting area with margin debug

### PR DESCRIPTION
**Pull request by 1222-takeshi**
## Description

One of the problems with the current intersection module is that vehicles traveling outside the lanelet cannot be considered as recognized vehicles even if they are at risk of collision.
In the future, this will be solved by improving the performance of the prediction, but this will take time.
Therefore, we consider solving this problem by extending the lanelet used for recognition in the lateral direction.
In this PR, we have achieved visualization of the extended lanelet.
In addition modified it to have parameters for the extension in the right and left directions.
## Tests performed
Example: right : 0.5 m, left : 0.5 m extension
image

Example: right : 2.0 m, left : 0.5 m extension
image

https://user-images.githubusercontent.com/57553950/171788794-da9f9288-ddae-44a7-8e2e-fb90f5022464.mp4

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/behavior_velocity_planner/include/scene_module/intersection/scene_intersection.hpp [hunk 1]
```diff
@@ -111,8 +112,12 @@ class IntersectionModule : public SceneModuleInterface
     double intersection_velocity;  //! used for intersection passing time
     double intersection_max_acc;   //! used for calculating intersection velocity
     double detection_area_margin;  //! used for detecting objects in detection area
```
Comment (tkimura4) [inline planning/behavior_velocity_planner/include/scene_module/intersection/scene_intersection.hpp:114]: Is this variable still used?

File: planning/behavior_velocity_planner/src/scene_module/intersection/manager.cpp [hunk 2]
```diff
@@ -75,6 +75,8 @@ IntersectionModuleManager::IntersectionModuleManager(rclcpp::Node & node)
   ip.intersection_velocity = node.declare_parameter(ns + ".intersection_velocity", 10.0 / 3.6);
   ip.intersection_max_acc = node.declare_parameter(ns + ".intersection_max_accel", 0.5);
   ip.detection_area_margin = node.declare_parameter(ns + ".detection_area_margin", 0.5);
+  ip.detection_area_right_margin = node.declare_parameter(ns + ".detection_area_right_margin", 0.5);
+  ip.detection_area_left_margin = node.declare_parameter(ns + ".detection_area_left_margin", 0.5);
```
Comment (tkimura4) [inline planning/behavior_velocity_planner/src/scene_module/intersection/manager.cpp:79]: Could you update yaml file too?

* Note that there are two yaml files, in behavior_velocity_planner and planning_launch.

**Comment by tkimura4** [comment]
The intersection module works well.
image

File: map/lanelet2_extension/lib/utilities.cpp [hunk 3]
```diff
@@ -303,6 +303,65 @@ lanelet::ConstLineString3d getCenterlineWithOffset(
...
+{
+  // Get length of longer border
+  const double left_length = lanelet::geometry::length(lanelet_obj.leftBound());
+  const double right_length = lanelet::geometry::length(lanelet_obj.rightBound());
+  const double longer_distance = (left_length > right_length) ? left_length : right_length;
+  const int num_segments = std::max(static_cast<int>(ceil(longer_distance / resolution)), 1);
+
+  // Resample points
+  const auto left_points = lanelet::utils::resamplePoints(lanelet_obj.leftBound(), num_segments);
+  const auto right_points = resamplePoints(lanelet_obj.rightBound(), num_segments);
```
Comment (tkimura4) [inline map/lanelet2_extension/lib/utilities.cpp:317]: lanelet::utils::resamplePoints -> resamplePoints?

File: map/lanelet2_extension/lib/utilities.cpp [hunk 4]
```diff
@@ -303,6 +303,65 @@ lanelet::ConstLineString3d getCenterlineWithOffset(
...
```
Comment (tkimura4) [inline map/lanelet2_extension/lib/utilities.cpp:346]: lanelet::utils::resamplePoints -> resamplePoints?

Comment (1222-takeshi) [reply map/lanelet2_extension/lib/utilities.cpp]: fix https://github.com/autowarefoundation/autoware.universe/pull/1021/commits/82570eaefae986ff2d48ca95b57877aaa16fc2c6

File: map/lanelet2_extension/lib/utilities.cpp (hunk 3 above)
Comment (1222-takeshi) [reply map/lanelet2_extension/lib/utilities.cpp]: fix https://github.com/autowarefoundation/autoware.universe/pull/1021/commits/82570eaefae986ff2d48ca95b57877aaa16fc2c6

File: planning/behavior_velocity_planner/src/scene_module/intersection/manager.cpp (hunk 2 above)
Comment (1222-takeshi) [reply planning/behavior_velocity_planner/src/scene_module/intersection/manager.cpp:79]: update yaml file https://github.com/autowarefoundation/autoware.universe/pull/1021/commits/35cb232d40f3a96d9d37fc5ada09b21ea72f1cd3

File: planning/behavior_velocity_planner/include/scene_module/intersection/scene_intersection.hpp (hunk 1 above)
Comment (1222-takeshi) [reply planning/behavior_velocity_planner/include/scene_module/intersection/scene_intersection.hpp:114]: use it here
https://github.com/autowarefoundation/autoware.universe/blob/4c910acd059e566c433ff3fd9bb9d83e1daf1e64/planning/behavior_velocity_planner/src/scene_module/intersection/scene_intersection.cpp#L608

**Review by tkimura4 — APPROVED** [review]
LGTM

**Pull request merged**